### ISSUE TITLE
Add more granual control to edit tracking

### ISSUE THREAD

**Issue by jamesbt365** [serenity-rs/poise#309]
Currently, any message that is edited (regardless of how old) is tracked if edit tracking is enabled. This can lead to users editing very old messages and it invoking a bot command.

**Comment by oinkrr** [comment]
I agree. It enables trolling by editing really old messages. I have already experienced that issue inside one of our servers where people edited messages from 2016.

**Comment by jamesbt365** [comment]
With discord sending the whole message payload on message edit now (instead of just the changed parts) this is more important than ever, the library should also include a way to ONLY edit messages that are cached, as discord will update old messages with embeds and some stuff sometimes.

Not sure of the specifics, but this does happen and may send responses into chat without any user interaction.

**Comment by jamesbt365** [comment]
This should be a decent priority because discord will occasionally send edit events for messages that were not edited by the user (i think its something to do with embeds) which are done in bulk over the channel which can lead to a bunch of command invocations that the user did not intend to trigger.
